### PR TITLE
Update replicated-sdk-overview.md

### PR DESCRIPTION
**Pull request by dexhorthy**
needs review from an SME, probably @marccampbell or @AmberAlston to ensure this is indeed the right direction we want to take things

### CODE REVIEW

**Comment by paigecalvert** [comment]
Thanks @dexhorthy . I added this story to our backlog for documenting this: https://app.shortcut.com/replicated/story/84712

**Comment by paigecalvert** [comment]
Also, pinged Amber and Marc here to give them a heads up that their input is requested: https://replicated.slack.com/archives/C02BPJQQP28/p1691072503053779 🙂

File: docs/vendor/replicated-sdk-overview.md [hunk 1]
```diff
@@ -123,4 +123,15 @@ For more information about viewing customer and instance insights in the vendor
 
 You can run the Replicated SDK in integration mode to more quickly test new functionality for your application. Integration mode allows you to use mock data to test changes locally without having to create a release in the vendor portal and then install your Helm chart from the Replicated registry.
 
-For more information, see [Developing Against the SDK API (Beta)](replicated-sdk-development).
\ No newline at end of file
+For more information, see [Developing Against the SDK API (Beta)](replicated-sdk-development).
+
+## Using an SDK-enabled Chart in a KOTS Release
+
+Many vendors will find it convenient to use the same Helm Chart for KOTS releases that they use for Helm CLI releases, in some cases even exposing the same Replicated release to both sets of customers. However, running both KOTS and the SDK can cause duplication of instance data in the Vendor Portal and related APIs. To avoid this, you will need to create a single Helm Chart that contains the Replicated SDK, but with a conditional value to exclude the SDK when the chart is being deployed by KOTS. To do this, you will need to make three changes to your application.
```
Comment (marccampbell) [inline docs/vendor/replicated-sdk-overview.md:130]: Why "single helm chart". I think this could be more accurately described as:

To avoid this, any of your Helm Charts that contain the Replicated SDK as a depedency or subchart will need to contain a conditional value to exclude the SDK...

Comment (dexhorthy) [inline docs/vendor/replicated-sdk-overview.md:130]:
```suggestion
Many vendors will find it convenient to use the same Helm Chart for KOTS releases that they use for Helm CLI releases, in some cases even exposing the same Replicated release to both sets of customers. However, running both KOTS and the SDK can cause duplication of instance data in the Vendor Portal and related APIs. To avoid this, any of your Helm Charts that contain the Replicated SDK will need a conditional value to exclude the SDK when the chart is being deployed by KOTS. To do this, you will need to make three changes to your application.
```

File: docs/vendor/distributing-workflow.md [hunk 2]
```diff
@@ -33,7 +31,12 @@ For more information about creating releases, see [Managing Releases with the Ve
   <tr>
     <td>HelmChart</td>
     <td><p>Provides instructions for KOTS about how to deploy your Helm chart.</p><p><strong>Note:</strong> Required for supporting KOTS installations of Helm charts.</p></td>
-    <td><a href="helm-native-v2-using">Configuring the HelmChart Custom Resource</a></td>
+    <td>
+      <ul>
+        <li><a href="helm-native-v2-using">Configuring the HelmChart Custom Resource</a></li>
+        <li><a href="helm-kots-using-sdk">Using an SDK-Enabled Helm Chart for KOTS Installations</a></li>
```
Comment (paigecalvert) [inline docs/vendor/distributing-workflow.md:37]: Add new topic ^

File: docs/vendor/helm-kots-using-sdk.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# Using an SDK-Enabled Chart for KOTS Installations
```
Comment (paigecalvert) [inline docs/vendor/helm-kots-using-sdk.md:1]: It felt like there was enough here to expand on this in it's own topic. I put it over in the "Helm Charts with KOTS" section of the KOTS sidebar, which is where we have other info about configuring the HelmChart custom resource.

I added links to this topic in a few places

File: docs/vendor/replicated-sdk-overview.md [hunk 4]
```diff
@@ -123,4 +123,10 @@ For more information about viewing customer and instance insights in the vendor
 
 You can run the Replicated SDK in integration mode to more quickly test new functionality for your application. Integration mode allows you to use mock data to test changes locally without having to create a release in the vendor portal and then install your Helm chart from the Replicated registry.
 
-For more information, see [Developing Against the SDK API (Beta)](replicated-sdk-development).
\ No newline at end of file
+For more information, see [Developing Against the SDK API (Beta)](replicated-sdk-development).
+
+## SDK-Enabled Helm Charts with KOTS
```
Comment (paigecalvert) [inline docs/vendor/replicated-sdk-overview.md:128]: I could also see this being called "Limitations", but as we only have one limitation right now, I think this heading does make the info easier to find. I could see people wondering "how's this thing work with kots?"

File: docs/vendor/helm-kots-using-sdk.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+   **Example:**
+
+    ```yaml
+    # Helm chart values.yaml
+    
+    replicated:
+      # Set `enabled: true` to include the SDK by default
+      enabled: true
+    ```
```
Comment (paigecalvert) [inline docs/vendor/helm-kots-using-sdk.md:47]: I thought it would be more descriptive to call the field `replictaed-sdk` rather than just `replicated`, but since the registry automatically injects a `replicated-sdk` field into the values file during install, I wasn't sure if that would cause issues. So, left as `replicated`

File: docs/vendor/distributing-workflow.md [hunk 6]
```diff
@@ -9,9 +9,7 @@ If you are new to Replicated, complete the following prerequisites before you ge
 * Complete a basic Replicated onboarding workflow to create an application and then promote and install initial releases in a development environment: 
   * (Recommended) For Helm chart-based applications, see [Replicated Quick Start](/vendor/replicated-onboarding).
   
-    :::note
     Distributing your application as a Helm chart is recommended because you can support both installations with the Helm CLI and with KOTS from the same release, without having to maintain separate sets of Helm charts or application manifests. For more information, see [Helm Charts with KOTS](/vendor/distributing-overview#helm-charts-with-kots) in _About Distributing Applications with Replicated_.
-    :::
```
Comment (paigecalvert) [inline docs/vendor/distributing-workflow.md:13]: ^ unrelated change, just found the note styling to be superfluous

File: docs/vendor/helm-kots-using-sdk.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+If you include the Replicated SDK as a dependency in a Helm chart that you want to use for both KOTS and Helm CLI installations, then you must add a conditional value to exclude the SDK when the chart is installed by KOTS. Excluding the SDK for KOTS installations prevents duplication of instance data in the vendor portal and related APIs, which can be caused when both KOTS and the SDK are reporting data for an instance.
+
+## Limitation
+
+When the SDK is excluded from KOTS installations, the SDK API endpoints are not available. For example, you cannot use the `app` endpoints to allow your users to check for updates when the SDK is excluded. And, you cannot use the `license` endpoints to check customer entitlements during runtime.
+
+However, KOTS also provides access to this functionality that is available through the SDK API. For example, users can check for updates through the admin console, and vendors can check license entitlements at runtime using the admin console API. If you use any SDK API endpoints in your application, Replicated recommends that you verify that the comparable functionality available through KOTS provides the expected user experience.
```
Comment (dexhorthy) [inline docs/vendor/helm-kots-using-sdk.md:15]: > vendors can check license entitlements at runtime using the admin console API

 this might not be as true as I thought it was

File: docs/vendor/helm-kots-using-sdk.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+## Overview
+
+Distributing your application as one or more Helm charts is recommended because you can use the same Helm charts to support installations with both KOTS and with the Helm CLI.
+
+If you include the Replicated SDK as a dependency in a Helm chart that you want to use for both KOTS and Helm CLI installations, then you must add a conditional value to exclude the SDK when the chart is installed by KOTS. Excluding the SDK for KOTS installations prevents duplication of instance data in the vendor portal and related APIs, which can be caused when both KOTS and the SDK are reporting data for an instance.
+
+## Limitation
+
+When the SDK is excluded from KOTS installations, the SDK API endpoints are not available. For example, you cannot use the `app` endpoints to allow your users to check for updates when the SDK is excluded. And, you cannot use the `license` endpoints to check customer entitlements during runtime.
```
Comment (dexhorthy) [inline docs/vendor/helm-kots-using-sdk.md:13]: we could add a note that we have on our roadmap (no guarantee if we'll do it or when) that we'd like to consolidate these APIs so that the same endpoints are available in KOTS as well.

@marccampbell @AmberAlston another ping on this question because IDK how publicly we want to discuss that, or if it's still the intended direction

**Review by jonquil2002 — DISMISSED** [review]
lgtm

File: docs/vendor/helm-kots-using-sdk.md (hunk 7 above)
Comment (AmberAlston) [reply docs/vendor/helm-kots-using-sdk.md:15]: We don't mention "admin console api" anywhere in our docs today because the kots api is only for kots internal use. We don't currently document or support that api for external use cases.

File: docs/vendor/helm-kots-using-sdk.md (hunk 8 above)
Comment (AmberAlston) [reply docs/vendor/helm-kots-using-sdk.md:13]: will look at updated content again, but we shouldn't mention anything that is not true today. We'd previously purposely stripped out any "future promises" from the docs. It's not a best practice, and there were several examples of poorly aged unfulfilled promises in the old kots docs. If and when we start to make this change, then we will update the docs accordingly.

**Comment by dexhorthy** [comment]
lgtm

File: docs/vendor/helm-kots-using-sdk.md (hunk 7 above)
Comment (AmberAlston) [reply docs/vendor/helm-kots-using-sdk.md]: That documented ability to query the entitlements is really our one exception to that. I wouldn't remove the content, but we may want to slightly tweak the language on that page a bit so it doesn't as strongly imply a supported kots api. More like "send an HTTP request to the admin console {license} endpoint at..."

Comment (paigecalvert) [reply docs/vendor/helm-kots-using-sdk.md]: Okay, that makes sense. I'll rewrite this limitation accordingly as well

Comment (paigecalvert) [reply docs/vendor/helm-kots-using-sdk.md]: Updated the language in this section to link to the related KOTS docs and remove mention of the admin console API

**Comment by paigecalvert** [comment]
Going to merge this in. Here's the follow up story for updating the language in the Checking Entitlements for KOTS topic per the discussion above about the admin console API: https://app.shortcut.com/replicated/story/85005
